modules/background-task/ios/ReactNativeBackgroundTask.mm

```diff
@@ -0,0 +1,109 @@
+#import "ReactNativeBackgroundTask.h"
+#import <UIKit/UIKit.h>
+#import <BackgroundTasks/BackgroundTasks.h>
+#import "RNBackgroundTaskManager.h"
+
+@implementation ReactNativeBackgroundTask {
+    NSMutableDictionary *_taskExecutors;
+}
+
+RCT_EXPORT_MODULE()
+
+- (instancetype)init {
+    if (self = [super init]) {
+        _taskExecutors = [NSMutableDictionary new];
+    }
+    return self;
+}
+
+- (BOOL)scheduleNewBackgroundTask:(NSString *)identifier {
+    BGAppRefreshTaskRequest *request = [[BGAppRefreshTaskRequest alloc] initWithIdentifier:identifier];
+    
+    // Set earliest begin date to some time in the future
+    request.earliestBeginDate = [NSDate dateWithTimeIntervalSinceNow:15 * 60]; // 15 minutes from now
+    
+    NSError *error = nil;
+    BOOL success = [[BGTaskScheduler sharedScheduler] submitTaskRequest:request error:&error];
+    
+    if (!success) {
+        NSLog(@"[ReactNativeBackgroundTask] Failed to schedule task: %@", error.localizedDescription);
+    }
+    
+    return success;
+}
+
+RCT_EXPORT_METHOD(defineTask:(NSString *)taskName
+                  taskExecutor:(RCTResponseSenderBlock)taskExecutor
+                  resolve:(RCTPromiseResolveBlock)resolve
+                  reject:(RCTPromiseRejectBlock)reject)
+{
+    if (taskName == nil) {
+        NSLog(@"[ReactNativeBackgroundTask] Failed to define task: taskName is nil");
+        reject(@"ERR_INVALID_TASK_NAME", @"Task name must be provided", nil);
+        return;
+    }
+    
+    if (taskExecutor == nil) {
+        NSLog(@"[ReactNativeBackgroundTask] Failed to define task: taskExecutor is nil");
+        reject(@"ERR_INVALID_TASK_EXECUTOR", @"Task executor must be provided", nil);
+        return;
+    }
+    
+    NSArray *backgroundIdentifiers = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"BGTaskSchedulerPermittedIdentifiers"];
+    
+    
+    if (!backgroundIdentifiers || ![backgroundIdentifiers isKindOfClass:[NSArray class]]) {
+        NSLog(@"[ReactNativeBackgroundTask] No background identifiers found or invalid format");
+        reject(@"ERR_INVALID_TASK_SCHEDULER_IDENTIFIER", @"No background identifiers found or invalid format", nil);
+        return;
+    }
+    
+    NSLog(@"[ReactNativeBackgroundTask] Defining task: %@", taskName);
+    
+    BOOL allSuccess = YES; 
+
+    for (NSString *identifier in backgroundIdentifiers) {
+        [[RNBackgroundTaskManager shared] setHandlerForIdentifier:identifier completion:^(BGTask * _Nonnull task) {
+            NSLog(@"[ReactNativeBackgroundTask] Executing background task's handler");
```

Comment:
In my previous experiences with background handling, we had some sort of expiry handling/time out logic. do you think we need it here?

Reply:
We could add a way for force finishing background task programmatically but for our use case I don't think that's necessary.